prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -370,4 +336,78 @@ class FHIRRouter(
                 ?: emptyList()
             ).plus(receiver.processingModeFilter)
     }
+
+    /**
+     * The list of FHIR Path variables to use for filters.
+     */
+    private val filterSubstitutionList: List<List<String>> = getFilterVarTable()
+
+    /**
+     * The list of FHIR Path variables to use for filters as a map.
+     */
+    private val filterSubstitutionMap: MutableMap<String, String>
+        get() {
+            return filterSubstitutionList.associate { it[0] to it[1] }.toMutableMap()
+        }
+
+    internal fun getFilterVarTable(): List<List<String>> {
+        val orderedList = mutableListOf<List<String>>()
+        val lookupTable = metadata.findLookupTable(filterVarTableName)
+        if (lookupTable != null && lookupTable.rowCount > 0) {
+            if (!lookupTable.hasColumn(filterVarTableNameCol) || !lookupTable.hasColumn(filterVarTablePathCol))
+                throw IllegalStateException(
+                    "Table $filterVarTableName is missing columns $filterVarTableNameCol and/or $filterVarTablePathCol"
+                )
+            // Extract all the rows
+            lookupTable.table.forEach { row ->
+                // Check for invalid names
+                val name = row.getString(filterVarTableNameCol)
+                if (filterVarNameFormatRegex.matches(name)) {
+                    orderedList.add(listOf(name, row.getString(filterVarTablePathCol)))
+                } else {
+                    logger.error(
+                        "Filter FHIR Path variable '$name' contains invalid characters. " +
+                            "Must match regex ${filterVarNameFormatRegex.pattern}"
+                    )
+                    // Do we completely fail here?
+                }
+            }
+            // Now put them in order of variable length, so short names variables do not overwrite longer ones
+            // that may contain the same name (e.g. patientAddress and patient, so we do not replace patient first).
+            orderedList.sortByDescending { it[0].length }
+            orderedList.toList()
+        } else {
+            logger.warn("Lookup table $filterVarTableName does not exist or is empty")
+        }
+
+        return orderedList
+    }
+
+    internal fun replaceFilterVars(path: String): String {
+        var processedPath = path
+
+        // We need to do recursive substitution, so keep substituting until the path no longer changes
+        // while (lastPathProcessed != processedPath) {
+        do {
+            val previousPath = processedPath
```

Comment:
Need a limiter here or we could get into some infinite loop due to self referencing variables